chord_metadata_service/chord/workflows/wdls/experiments_extended_json.wdl

```diff
@@ -0,0 +1,404 @@
+version 1.0
+
+workflow experiments_extended_json {
+    input {
+        String directory
+        File json_document
+        String drs_url
+        String katsu_url
+        String project_dataset
+        String access_token
+        Boolean validate_ssl
+    }
+
+    call prepare_download_list {
+        input:
+            json_document = json_document,
+            directory = directory
+    }
+
+    ## Calculate the size of the JSON document to determine if there are any URLs to download
+    Int urls_json_size = ceil(size(prepare_download_list.urls_json, "Gi") * 8000)
+
+    if (urls_json_size > 0) {
+        call process_downloads {
+            input:
+                urls_json = prepare_download_list.urls_json
+        }
+
+        call prepare_for_download {
+            input:
+                json_document = prepare_download_list.urls_json
+        }
+
+        scatter (download_info in prepare_for_download.download_list) {
+            call download_file {
+                input:
+                    url = download_info.url,
+                    filename = download_info.filename,
+                    directory = directory,
+                    access_token = access_token,
+                    validate_ssl = validate_ssl
+            }
+        }
+    }
+
+    call prepare_for_drs {
+        input:
+            json_download_list = prepare_download_list.urls_json,
+            json_path_list = prepare_download_list.path_list,
+            downloaded_files = download_file.downloaded_file
```

Comment:
Does this pass all the downloaded files? Seems to me like it would be only the one from the last `download_file` call

Reply:
I think scatter turns this into an array - WDL is weird

Reply:
Yes, files are implicitly collected into an array because they are generated within a scatter block